x/feegrant/migrations/v045/store.go

```diff
@@ -0,0 +1,50 @@
+package v045
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/feegrant"
+)
+
+func addAllowancesByExpTimeQueue(ctx types.Context, store storetypes.KVStore, cdc codec.BinaryCodec) error {
+	prefixStore := prefix.NewStore(store, feegrant.FeeAllowanceKeyPrefix)
+	iterator := prefixStore.Iterator(nil, nil)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		var grant feegrant.Grant
+		bz := iterator.Value()
+		if err := cdc.Unmarshal(bz, &grant); err != nil {
+			return err
+		}
+
+		grantInfo, err := grant.GetGrant()
+		if err != nil {
+			return err
+		}
+
+		exp, err := grantInfo.ExpiresAt()
+		if err != nil {
+			return err
+		}
+
+		if exp != nil {
+			key := iterator.Key()
+			if exp.Before(ctx.BlockTime()) {
+				prefixStore.Delete(key)
+			} else {
+				grantByExpTimeQueueKey := feegrant.FeeAllowancePrefixQueue(exp, key)
```

Comment:
In general, I prefer to keep these key functions inside the v045 migration package. This way, if we change it one day in the main feegrant logic, we won't run into bugs for old migrations